packages/scaffold-config/test/unit/detect.spec.ts

```diff
@@ -1,100 +1,199 @@
 import { expect } from 'chai'
+import fs from 'fs-extra'
 import { e2eProjectDirs } from '@packages/frontend-shared/cypress/e2e/support/e2eProjectDirs'
 import { detect } from '../../src'
 import Fixtures from '@tooling/system-tests/lib/fixtures'
 import path from 'path'
-import fs from 'fs-extra'
 
 export async function scaffoldMigrationProject (project: typeof e2eProjectDirs[number]) {
   Fixtures.removeProject(project)
 
   await Fixtures.scaffoldProject(project)
 
-  return fs.readJSONSync(path.join(Fixtures.projectPath(project), 'package.json'))
+  return path.join(Fixtures.projectPath(project))
+}
+
+interface DepToFake {
+  dependency: string
+  version: string
+}
+
+interface DevDepToFake {
+  devDependency: string
+  version: string
+}
+
+/**
+ * The way we detect dependencies is by using resolve-from (https://www.npmjs.com/package/resolve-from).
```

Comment:
I wonder if we can just do `require.resolve`? We already were using `resolve-from`, but I don't think it seems to do anything special `require.resolve` doesn't?

In the README it says

> Resolve the path of a module like [require.resolve()](https://nodejs.org/api/globals.html#globals_require_resolve) but from a given pathResolve the path of a module like [require.resolve()](https://nodejs.org/api/globals.html#globals_require_resolve) but from a given path

I think we can just do `require.resolve('react-scripts', { paths: [this.ctx.currentProject] })`.

@elevatebart I see you added the `resolve-from` module a while ago, did you know about `require.resolve(m, { paths: ... })`? Or did I miss something about what `resolve-from` does?

Reply:
If tests pass with `require.resolve()` you should use that instead. Less dependencies is always better.
At the time, I had trouble with refreshing the cached values while running the app. A dependency that would not be there when the process started would never be detected as added when installed.

Do you want me to try and remove `resolve-from` in a separate PR?

Reply:
Sure, how about after this one lands? Then you'll have the test coverage to refactor with more confidence.